### ISSUE TITLE
Loadbalaning issues with traefik and Docker Swarm cluster

### ISSUE THREAD

**Issue by prateek2408** [containous/traefik#730]
Hi,

I am using the latest traefik binary (v1.1.0-rc1) :-
![traefik](https://cloud.githubusercontent.com/assets/16288331/19312115/b1550536-90ae-11e6-96c6-49c274161a83.JPG)

Docker swarm which I am using is -  
![swarm](https://cloud.githubusercontent.com/assets/16288331/19312712/4d1a8fe8-90b1-11e6-934f-48e18f787aba.JPG)

I performed the following activity:- 
1. Create a swarm cluster of 7 nodes.
2. Created an overlay network.
3. Created a traefik container using docker service create
4. Created 4 instances of 'emilevauge/whoami' via docker service create and docker service scale <name>=4

Now if I go into the traefik container and do a curl on the whoami instance IP i get different host names signifying that the load is getting balanced, here are the snippets:-

![snip1](https://cloud.githubusercontent.com/assets/16288331/19312374/bc563814-90af-11e6-8861-2b3dafe948eb.JPG)
![snip2](https://cloud.githubusercontent.com/assets/16288331/19312373/bc54e400-90af-11e6-82c2-bc9223a314b8.JPG)
![snip3](https://cloud.githubusercontent.com/assets/16288331/19312375/bc62e118-90af-11e6-83ba-3a9cb6f2d866.JPG)
 
However the traefik binary which is running in the same container and reaching the whoami service via the same internal IP is giving me same hostnames telling that load is going only to a single container. 

Seems it is stuck to a single instance. 

In the traefik logs I see that it is redirecting internally to the same IP so ideally I should get a different hostname, moreover load balancing is now taken care by swarm and has nothing to do with traefik but still this issue is coming for me when I am using traefik . Here is the snippet of traefik log

![traefik round](https://cloud.githubusercontent.com/assets/16288331/19312592/c005b524-90b0-11e6-8968-59e33c9e7f7f.JPG)

I am clueless here... Can anyone help.

Regards,
Prateek

**Comment by mthird** [comment]
Are you using DNSRR or VIP mode for your backend service?

**Comment by emilevauge** [comment]
@prateek2408 could you give us your traefik config (TOML, args)?

**Comment by prateek2408** [comment]
@mthird - Nopes I am just using plain Swarm cluster and under that I created a service and then scaled it to 4 instances, goes something like this: -

1. docker network create --driver overlay mynet
2. docker service create --network mynet emilevauge/whoami .....<all parameters>
3. docker service scale <service-id>=
4. docker service inspect <service-id>

Now the service inspect output will give me a single IP from the overlay network. If i internally do a curl to this IP( screenshots attached above) I get different hostnames, but when traefik uses it same hostnames are being displayed signifying that load balancing of swarm is not working

@emilevauge - I ran traefik binary in the container in the following way -

./traefik_linux-amd64 --web --docker.swarmmode --docker.watch

It runs and as the container on which it is running has docker unix socket shared so it connects to the docker service and starts listening on port 8080 for WEBUI and 80 for front end.

**Comment by vovimayhem** [comment]
I've also noticed this.

The cluster was created following the [v1.1 Swarm Mode Guide](https://github.com/containous/traefik/blob/v1.1/docs/user-guide/swarm-mode.md) with some minor corrections, as I was deploying on Azure.

This is how I created the Traefik service on the swarm (I changedjust the network name and the domain when I ran this command):
```
docker service create \
    --name traefik \
    --constraint 'node.role==manager' \
    --publish 80:80 \
    --publish 8080:8080 \
    --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
    --network traefik-net \
    traefik:camembert \
    --docker \
    --docker.swarmmode \
    --docker.domain=traefik \
    --docker.watch \
    --logLevel=DEBUG \
    --web
```

Then this is how I created the service:
```
docker service create \
  --name whoami0 \
  --label 'traefik.port=80' \
  --network traefik-net \
  emilevauge/whoami
```

And this is how I scaled the service:

```
docker service scale whoami0=3
```

Tested with the following command:
```
curl -H Host:whoami0.traefik http://$(docker-machine ip manager)
```

Instead of seeing each request being processed by each of the three containers (a round robin), it just sticked with one.

**Comment by vovimayhem** [comment]
And it looks like the sticky behavior is at traefik level. If I try curl-ing the site after a few minutes, another container starts processing.

**Comment by sunnysideocean** [comment]
I tried reproducing this problem but the load balancing works fine for me under the swarm mode setup. I tested it in swarm mode with labels (in which case the IP is added to traefik). I also tested it using a file config where I added the host name of the container.

However, I noticed that the traffic is first redirected to container A and after many requests, traffic is routed to container B for a while. So with a load test I can see that the traffic switches between the 2 containsers every few seconds (~7 seconds in my test).

I did the same setup with nginx as reverse proxy and there the traffic was distributed evenly between the containers at all time.

Nevertheless, eben with traefik the load balancing works - altough it happens in batches and of course if may work due to the docker ingress load balancing.
